lib/readline.js

```diff
@@ -238,6 +243,24 @@ Interface.prototype.setPrompt = function(prompt) {
   this._prompt = prompt;
 };
 
+Interface.prototype.read = function() {
+  if (this._buffer.length === 0)
+    return null;
+
+  return this._buffer.shift();
+};
+
+
+Interface.prototype[Symbol.asyncIterator] = function() {
+  emitExperimentalWarning('readline Interface[Symbol.asyncIterator]');
+
+  this._enableBuffer = true;
+  this.close();
+  this.input.setEncoding('utf8');
```

Comment:
It's definitely not super friendly when a module like readline changes stream settings, especially when we only do it with async iteration. We should add this feature without it.